numba_dpex/kernel_api/atomic_ref.py

```diff
@@ -200,3 +200,55 @@ def exchange(self, val):
         old = self._ref[self._index]
         self._ref[self._index] = val
         return old
+
+    def compare_exchange_weak(self, expected, desired, expected_idx=0):
+        """Compares the value of the object referenced by the AtomicRef
+        against the value of ``expected[expected_idx]``.
+        If the values are equal, attempts to replace the value of the
+        referenced object with the value of ``desired``.
+        Otherwise assigns the original value of the
+        referenced object to ``expected[expected_idx]``.
+
+        Args:
+            expected : Vector containing the expected value of the
+            object referenced by the AtomicRef.
+            desired : Value that replaces the value of the object
+            referenced by the AtomicRef.
+            expected_idx: Offset in `expected` vector where the expected
+            value of the object referenced by the AtomicRef is present.
+
+        Returns: Returns ``True`` if the comparison operation and
+        replacement operation were successful.
+
+        """
+        if self._ref[self._index] == expected[expected_idx]:
+            self._ref[self._index] = desired
+            return True
+        expected[expected_idx] = self._ref[self._index]
+        return False
+
+    def compare_exchange_strong(self, expected, desired, expected_idx=0):
```

Comment:
What is the difference between the two variants of `CAS`? When should one use one over the other?

Reply:
Perhaps we should just have one variant?